app/models/solr_document.rb

```diff
@@ -225,35 +224,45 @@ def formatted_page_text
     return @formatted_page_text
   end
 
-  def get_actual_txt_file
-    base_name="https://stacks.stanford.edu/file/druid:#{self.druid}/"     
-    if self[:ocr_id_ss].blank?
-      @formatted_page_text=""
-      @txt_file=""
+  # the ocr file on the stacks can unfortunately be named by various possibilities, so we have to look for each
+  def possible_ocr_filenames
+    ocr_id=self[:ocr_id_ss]
+    if ocr_id.blank?
+      []
     else
-      possible_filenames=[self[:ocr_id_ss],self[:ocr_id_ss].gsub('_99_','_'),self[:ocr_id_ss].gsub('_99_','_00_')]
-      possible_filenames.each do |file| 
-        begin
-           full_path="#{base_name}#{file}"
-           @formatted_page_text=open(full_path).read.encode('UTF-16le', :invalid => :replace, :replace => '').encode('UTF-8')
-           @txt_file=full_path
-           break
-         rescue
-           @formatted_page_text=""
-           @txt_file=""
-         end
-      end
+      [ocr_id,ocr_id.gsub('_99_','_'),ocr_id.gsub('_99_','_00_')]
     end
   end
-  
+
+  def get_actual_txt_file
+    base_name="#{Frda::Application.config.stacks_url}/file/druid:#{self.druid}/"
+    @formatted_page_text=""
+    @txt_file=""
+    possible_ocr_filenames.each do |file|
+      txt_file="#{base_name}#{file}"
+      logger.info("app/models/solr_document.rb#get_actual_txt_file: Looking for OCR file #{txt_file}")
+      begin
+        response = Faraday.get(txt_file)
+        if response.status == 200 # we found it
+          logger.info("....found #{txt_file}")
+          @txt_file=txt_file # cache the filename
+          @formatted_page_text=response.body.force_encoding('UTF-8').scrub # encoding of txt files from the stacks is set to ASCII-8BIT, but is REALLY UTF-8, force it
+          break # don't bother checking for more filename possibilities once we find one
+        end # end check for 200 response code
+      rescue Exception => e
```

Comment:
Irrespective of whether or not to use a blanket `rescue` here, when you do need this, you should rescue `StandardError` instead of `Exception`. Basically, you should never rescue `Exception`. That seems weird, but Ruby's exception hierarchy is weird -- see here for an explanation: https://robots.thoughtbot.com/rescue-standarderror-not-exception